Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Layouts.cs

```diff
@@ -1853,6 +1853,63 @@ public void Layouts_CanSetUpMultipleLayoutsFromSingleState_UsingVariants()
         Assert.That(deviceB["axis"].variants, Is.EqualTo("B"));
     }
 
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_ParseParameters_EmptyStringReturnsNull()
+    {
+        var parameters = InputControlLayout.ParseParameters("");
+        Assert.AreEqual(null, parameters, "Empty string should generate null parameters");
+    }
+
+    void Layouts_ParseParameters_GenericWorks<T>(T[] expectedValues, InputControlLayout.ParameterType parameterType, string input) where T : unmanaged
+    {
+        var parameters = InputControlLayout.ParseParameters(input);
+        Assert.AreEqual(expectedValues.Length, parameters.Length, "Number of parameters does not match.");
+        unsafe
+        {
+            for (int i = 0; i < expectedValues.Length; i++)
+            {
+                var p = parameters[i];
+                Assert.AreEqual(parameterType, p.type, $"Unexpected type in parameter {i}");
+                Assert.AreEqual($"p{i}", p.name, $"Unexpected name in parameter {i}");
+                Assert.AreEqual(expectedValues[i], *(T*)p.value, $"Unexpected value in parameter {i}");
+            }
+        }
+    }
+
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_ParseParameters_BoolsWorks()
+    {
+        Layouts_ParseParameters_GenericWorks(
+            new bool[] { true, false, true, false },
+            InputControlLayout.ParameterType.Boolean,
+            "p0=true,p1=false,p2=TRUE,p3=False"
+        );
+    }
+
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_ParseParameters_IntWorks()
+    {
+        Layouts_ParseParameters_GenericWorks(
+            new int[] { 0, 0, 1234, -5678, 0 },
+            InputControlLayout.ParameterType.Integer,
+            "p0=0,p1=-0,p2=1234,p3=-5678,p4=12345678901234567890"
+        );
+    }
+
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_ParseParameters_FloatWorks()
```

Comment:
No `XYZWorks` style tests. Try to always formulate the test as a requirement where verifying that things work correctly is more or less a side effect and the test focuses on specifying an ability the system should have or trait it should exhibit.